test/type_prop.cpp

```diff
@@ -4343,12 +4343,32 @@ TEST(type_prop, one_hot_deduce_matrix_2)
     ASSERT_EQ(oh->get_shape(), (Shape{12, 24, 2}));
 }
 
+TEST(type_prop, one_hot_deduce_et_dynamic)
+{
+    auto param = make_shared<op::Parameter>(element::dynamic, Shape{12, 24});
+    auto oh = make_shared<op::OneHot>(param, Shape{12, 24, 2}, 2);
+    ASSERT_EQ(oh->get_element_type(), element::dynamic);
+    ASSERT_EQ(oh->get_shape(), (Shape{12, 24, 2}));
+}
+
 TEST(type_prop, one_hot_deduce_floating_point)
 {
     auto param = make_shared<op::Parameter>(element::f32, Shape{12, 24});
-    auto oh = make_shared<op::OneHot>(param, Shape{12, 24, 8}, 2);
-    ASSERT_EQ(oh->get_element_type(), element::f32);
-    ASSERT_EQ(oh->get_shape(), (Shape{12, 24, 8}));
+    try
+    {
+        auto oh = make_shared<op::OneHot>(param, Shape{12, 24, 8}, 3);
```

Comment:
Use EXPECT_THROW or whatever that call is. Your check is too strict in that it expects an exact string match. It should throw an `invalid_argument` exceptions since one of the arguments is invalid. Don't check for string match.

Reply:
Can we revisit this some other time? This pattern is the same for all 378 billion type prop tests, and I do think we need a way to be sure that a reasonably informative error string has been produced (which `EXPECT_THROW` doesn't seem to help with), since users are relying on these error messages.

Reply:
It is convenient that English is the only language used in the world, otherwise this could be difficult.

Reply:
These are type prop tests, not backend tests.